Taskfile.yml

```diff
@@ -245,13 +247,14 @@ tasks:
           DATA_DIR: "{{.OUTPUT_DIR}}"
     cmds:
       - "rm -rf '{{.OUTPUT_DIR}}'"
-      - for:
-          - "client"
-          - "yscope-log-viewer"
-        cmd: |-
-          cd "{{.ITEM}}"
-          PATH="{{.G_NODEJS_22_BIN_DIR}}":$PATH npm run build -- \
-          --output-path "{{.OUTPUT_DIR}}/{{.ITEM}}"
+      - |-
+        cd "client"
+        PATH="{{.G_NODEJS_22_BIN_DIR}}":$PATH npm run build -- \
+          --outDir "{{.OUTPUT_DIR}}/client"
```

Comment:
I assume --outDir is type script specific?

I don't mind you break the "for" into two separate lines, but just fyi we have a way to specify target-specific flags for each target in a "for". It might not worth the effort to do it here though

Reply:
do you mean we can write something like this instead?
```
      - for:
          - {"PATH": "client", "OUTPUT_DIR_FLAG": "--outDir"}
          - {"PATH": "yscope-log-viewer", "OUTPUT_DIR_FLAG": "--output-path"}
        cmd: |-
          cd "{{.ITEM.PATH}}"
          PATH="{{.G_NODEJS_22_BIN_DIR}}":$PATH npm run build -- \
          {{.ITEM.OUTPUT_DIR_FLAG}} "{{.OUTPUT_DIR}}/{{.ITEM.PATH}}"
```
if so, i agree that's cleaner

Reply:
Actually, what I wrote in the past was something else. 

What I did is that I factor out the entire command as a subtask, and let it consumes two variables:

one example can be found here. see the difference between py-check and py-fix. https://github.com/y-scope/clp-ffi-py/blob/main/lint-tasks.yml#L135

There might be a way to write what you suggested with "for", but I have personally never tried it.

Reply:
ah, what i proposed was actually valid syntax. I guess for now, the "for" approach can be better given that the extracted task might not be reusable in other tasks at the moment.

Reply:
applied the "for" approach for now. we can revert if we disagree.

Reply:
Sorry forgot to reply this. it looks good to me.